src/queries/a11yHint.ts

```diff
@@ -0,0 +1,68 @@
+import type { ReactTestInstance } from 'react-test-renderer';
+import { TextMatch } from '../matches';
+import { makeQueries } from './makeQueries';
+import type {
+  FindAllByQuery,
+  FindByQuery,
+  GetAllByQuery,
+  GetByQuery,
+  QueryAllByQuery,
+  QueryByQuery,
+} from './makeQueries';
+
+export function matchStringValue(
+  prop: string | undefined,
+  matcher: TextMatch
+): boolean {
+  if (!prop) {
+    return false;
+  }
+
+  if (typeof matcher === 'string') {
+    return prop === matcher;
+  }
+
+  return Boolean(prop.match(matcher));
+}
+
+const queryAllByA11yHint = (
+  instance: ReactTestInstance
+): ((displayValue: TextMatch) => Array<ReactTestInstance>) =>
+  function queryAllByDisplayValueFn(displayValue) {
+    return instance.findAll(
+      (node) =>
+        typeof node.type === 'string' &&
+        matchStringValue(node.props.accessibilityHint, displayValue)
+    );
+  };
+
+const getMultipleError = (a11yHint: TextMatch) =>
+  `Found multiple elements with accessibilityHint: ${String(a11yHint)} `;
+const getMissingError = (a11yHint: TextMatch) =>
+  `Unable to find an element with accessibilityHint: ${String(a11yHint)}`;
+
+const { getBy, getAllBy, queryBy, queryAllBy, findBy, findAllBy } = makeQueries(
+  queryAllByA11yHint,
+  getMissingError,
+  getMultipleError
+);
+
+export type ByA11yHintQueries = {
```

Comment:
@mdjastrzebski  Oh indeed I missed that! But in the previous API, when I look at the [interface](https://github.com/callstack/react-native-testing-library/blob/597fa3c4f84b40cccfd28a2babeb90d98dbe7c93/typings/index.flow.js#L214) `A11yAPI`  , I only see `getByHintText` and `getByA11yHint` but not `getByAccessibilityHint`. Should I add it then? 

In the same way, I don't see `getByAccessiblityState` nor `getByState` in the former API nor any of the other aliases for states, role, label etc... (byAccessibilityXXX or getByXXX). Should I add them all? I agree that it would be better if every query followed the same pattern.
If I do change the API, should I do in this PR or in another one? (to keep this one a pure refactoring)

Reply:
Long story short about a11y prop names:
1. originally we had all a11y queries named `getByA11yXxx` & `getByAccessibilityXxx`. 
2. At some stage we decided to use RTL naming for these queries which had direct counterparts in RTL (iirc: `byRole`) and keep the existing `byA11y` name for all other which are not present in RTL. 
3. However we also kept double naming for `byHintText`/`byA11yHint` because it was present in alternative RNTL library which we merged with.

Generally I would not remove or rename any queries, because that would cause a breaking change. We might might fill in some missing gaps as that is not breaking.